### ISSUE TITLE
Announcements about an organization, its services, and/or locations.

### ISSUE THREAD

**Issue by greggish** [openreferral/specification#216]
I've been asked about data elements to structure announcements that might be associated with an organization, location, or service. This is especially important during times of crisis, when information about service provision may change rapidly. We currently have the ability to indicate whether a service is 'active' or not, but could it be valuable to provide more context. This is different than a 'notes' field, in that it could officially come from the organization and might be time-limited.

I've seen users develop workarounds for this (at the extreme level: during a disaster many referral providers will abandon their structured resource data systems and instead start using  word processing documents, because their information systems can't handle the sudden variability of resource information) but I haven't seen instances of this represented in a data structure. That said, [google is proposing a `SpecialAnnouncements` element](https://developers.google.com/search/docs/data-types/special-announcements) to schema.org and it might make sense to develop a similar entity.

**Comment by robredpath** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Investigate the question of announcements](https://trello.com/c/DZ3BpNBg/11-investigate-the-question-of-announcements)

**Comment by Rovack** [comment]
This sounds like a feature that would be super helpful at times like these.

When building a COVID-adapted version of our own [HSDS-based API](https://github.com/streetlives/streetlives-api), we found we had to make a few adjustments/extensions to the schema along similar lines.
While we don't have an Announcements table per se, we did end up with some structures that might be relevant for this discussion, so maybe this will be useful:

For one, since we wanted to separate **COVID opening hours** from the the regular opening hours data - instead of overwriting it - we (ab)used the `holiday-schedules` table, with the addition of an `occasion` field (as inappropriate as calling this a "holiday" may be...).
It also required allowing `weekday` instead of `start_date`/`end_date`, as an occasion like this lasts for months and the end date is unknown. (See all changes [here](https://github.com/streetlives/streetlives-api/pull/29/files#diff-24d3fca3e64b596ef5466491c748b88b).)
In retrospect, maybe it would have made more sense to copy the `closed` flag from `holiday-schedules` to `regular-schedules` and add the `occasion` field to the `regular-schedules` table. Or just create a new table altogether.

More directly related to the notion of Announcements, we also added a [table](https://github.com/streetlives/streetlives-api/blob/develop/src/models/event-related-info.js) for "**Event-Related Info**", with just an `event` field along with the free-text `information`.
This has a many-to-one relationship to both `services` and `locations` (though currently there's a unique constraint for service+location+event, meaning for e.g. COVID it's effectively one-to-one).
At the Service level, we use it to represent any free-form textual info regarding changes to the service's operation during the pandemic. At the Location level, currently we only use it for locations that are entirely closed, where we just want to indicate info like e.g. when people can expect them to reopen, whom they can contact in the meanwhile, etc.

There's no direct relationship between this table and the special opening hours, except `event-related-info.event` and `holiday-schedules.occasion` use the same value.
I suppose for a less ad-hoc solution, it could make sense to either directly connect the special schedules to Announcements, or at least have an `events`/`occasions` table that's related to both.

**Comment by AtticusRains** [comment]
I agree with @Rovack about the need for the distinction between emergency hours and regular hours to accompany announcements; especially for shorter-term natural disasters where normal operations may resume in a matter of days or weeks instead of months as with this crisis. As I believe Rob touched on in a previous development call, this may be able to be handled by the more complex date-time representations currently being discussed (#167) but I believe guidance regarding this use case and its integration with announcements is important for consistency. Hunger Free America has also seen temporary changes to other tables such as eligibility that would be nice to highlight with an announcement implementation.

**Comment by greggish** [comment]
Here is some more context for the `SpecialAnnouncements` element via schema.org, which various parts of the federal gov seem to have implemented: 
https://digital.gov/2020/05/11/experimenting-with-specialannouncement-markup/

**Comment by markdav-is** [comment]
We have an "alerts" feature at https://littlehelpbook.com/ that allows for showing emergency info.  We also have an Infos entity that is a simplified service entity that can participate in queries, filters, catagories, etc.

**Comment by Cskyleryoung** [comment]
I just left a comment along these lines here: https://github.com/openreferral/specification/issues/221

In our search engine for Washington 211, we added an `emergency_info` field that gets displayed at the top of records. It's been extensively used during COVID.
